### PR TITLE
Update include_role description

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY

Rebased and updated version of #50204. 

Clarifies documentation of `include_role`: where is it allowed, etc.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs.ansible.com
include_role

### CODE REVIEW

File: lib/ansible/modules/utilities/logic/include_role.py [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
 short_description: Load and execute a role
 description:
-  - Loads and executes a role as a task dynamically.
-  - This frees roles from the C(roles:) directive and allows them to be treated more as tasks.
-  - Unlike M(import_role), most keywords, including loop, with_items, and conditionals, apply to this statement.
-  - The do until loop is not supported on M(include_role).
+  - Dynamically loads and executes a specified role as a task.
+  - May be used only where Ansible tasks are allowed - inside C(pre_tasks:), C(tasks:), or C(post_tasks:) playbook objects, or as a task inside a role.
+  - Supports most keywords, including C(loop), C(with_items), and C(conditionals) (unlike M(import_role)).
+  - Does not support C(do_until) loops.
```
Comment (bcoca) [inline lib/ansible/modules/utilities/logic/include_role.py]: `until:` is the proper keyword for those loops

File: lib/ansible/modules/utilities/logic/include_role.py [hunk 2]
```diff
@@ -19,16 +19,16 @@
 module: include_role
 short_description: Load and execute a role
 description:
-  - Loads and executes a role as a task dynamically.
-  - This frees roles from the C(roles:) directive and allows them to be treated more as tasks.
-  - Unlike M(import_role), most keywords, including loop, with_items, and conditionals, apply to this statement.
-  - The do until loop is not supported on M(include_role).
+  - Dynamically loads and executes a specified role as a task.
+  - May be used only where Ansible tasks are allowed - inside C(pre_tasks:), C(tasks:), or C(post_tasks:) playbook objects, or as a task inside a role.
+  - Supports most keywords, including C(loop), C(with_items), and C(conditionals) (unlike M(import_role)).
```
Comment (bcoca) [inline lib/ansible/modules/utilities/logic/include_role.py]: `import_role` supports those keywords, but for inheritance, for include it applies to the task itself, the original phrasing was more accurate

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/utilities/logic/include_role.py [hunk 3]
```diff
@@ -19,16 +19,16 @@
 module: include_role
 short_description: Load and execute a role
 description:
-  - Loads and executes a role as a task dynamically.
-  - This frees roles from the C(roles:) directive and allows them to be treated more as tasks.
-  - Unlike M(import_role), most keywords, including loop, with_items, and conditionals, apply to this statement.
-  - The do until loop is not supported on M(include_role).
+  - Dynamically loads and executes a specified role as a task.
+  - May be used only where Ansible tasks are allowed - inside C(pre_tasks), C(tasks), or C(post_tasks) playbook objects, or as a task inside a role.
+  - Task-level keywords, L(loops,../user_guide/playbooks_loops.html), and L(conditionals,../user_guide/playbooks_conditionals.html) will be applied only to the C(include_role) statement itself. If you want the opposite behavior, use M(import_role) instead.
```
Comment (bcoca) [inline lib/ansible/modules/utilities/logic/include_role.py]: 1/2 is good, the other 1/2 should be more. 'If you want to have keywords applied to the contained tasks, consider using  import_role instead or use the "apply" keyword'

File: lib/ansible/modules/utilities/logic/include_role.py [hunk 4]
```diff
@@ -19,16 +19,16 @@
...
+  - The C(until) keyword, which retries a task until a condition is met, does not apply to C(include_role).
```
Comment (bcoca) [inline lib/ansible/modules/utilities/logic/include_role.py]: I would rephrase since its not just `until` 'There are several keywords ignored by this action, like `until` and `retries`.

**Review by bcoca — CHANGES_REQUESTED** [review]

**Review by bcoca — APPROVED** [review]

**Review by samccann — APPROVED** [review]
